### PR TITLE
Implement storage

### PR DESCRIPTION
**Pull request by 8kdesign**
Split storage into two parts: Loader and Writer.

Interact with them through methods in ModuleList:
1) Use addModule to create a new module.
2) Use removeModule to delete a module.
3) Use setSelectedModule to select a module. Returns false if not found.
4) Use reset to set selected module to null.

To update change to current module, call writeModule in Writer.

### CODE REVIEW

File: src/main/java/seedu/duke/lesson/Lesson.java [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 public class Lesson {
 
-    private String time;
+    private String time = "";
     private final LessonType LESSON_TYPE;
-    private TeachingStaff teachingStaff;
-    private String onlineLink;
+    private TeachingStaff teachingStaff = new TeachingStaff("","");
+    private String onlineLink = "";
```
Comment (isaharon) [inline src/main/java/seedu/duke/lesson/Lesson.java:8]: Consider creating a default constructor

File: src/main/java/seedu/duke/lesson/TeachingStaff.java [hunk 2]
```diff
@@ -2,6 +2,19 @@
 
 public class TeachingStaff {
 
-    String name;
-    String email;
+    private final String name;
```
Comment (isaharon) [inline src/main/java/seedu/duke/lesson/TeachingStaff.java:5]: Remove final

File: src/main/java/seedu/duke/module/ModuleList.java [hunk 3]
```diff
@@ -1,25 +1,74 @@
...
+import seedu.duke.storage.Loader;
+import seedu.duke.storage.Writer;
 
 import java.util.ArrayList;
 
 public class ModuleList {
 
-    private static final Storage storage = new Storage();
+    private static final Loader loader = new Loader();
+    private static final Writer writer = new Writer();
```
Comment (isaharon) [inline src/main/java/seedu/duke/module/ModuleList.java:11]: To remove from object class

File: src/main/java/seedu/duke/storage/Loader.java [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    public void loadModuleNames() {
+        File directory = new File(Constants.FOLDER_PATH);
+        File[] files = directory.listFiles();
+        if (files == null) {
+            return;
+        }
+        for (File file : files) {
+            String name = file.getName();
+            if (name.endsWith(Constants.FILE_FORMAT)) {
+                ModuleList.addModule(name.replace(Constants.FILE_FORMAT, ""));
```
Comment (isaharon) [inline src/main/java/seedu/duke/storage/Loader.java:33]: addModule creates new file after loading?

File: src/main/java/seedu/duke/task/Task.java [hunk 5]
```diff
@@ -4,11 +4,11 @@
 
 public class Task {
 
-    private String description;
-    private String remarks;
+    private String description = "";
+    private String remarks = "";
```
Comment (isaharon) [inline src/main/java/seedu/duke/task/Task.java:8]: Consider creating a default constructor

**Review by isaharon — CHANGES_REQUESTED** [review]

**Review by isaharon — COMMENTED** [review]
LGTM!

**Review by isaharon — APPROVED** [review]

File: src/main/java/seedu/duke/storage/Loader.java [hunk 6]
```diff
@@ -0,0 +1,213 @@
+package seedu.duke.storage;
+
+import seedu.duke.common.Constants;
```
Comment (ivanchongzhien) [inline src/main/java/seedu/duke/storage/Loader.java:3]: Consider replacing this line with:

import static seedu.duke.common.Constants.(constant name)

so you don't have to repeatedly use Constants.(constant name)? I think if you are using most of the constants in that class, it is acceptable to use *.

File: src/main/java/seedu/duke/storage/Writer.java [hunk 7]
```diff
@@ -0,0 +1,212 @@
+package seedu.duke.storage;
+
+import seedu.duke.common.Constants;
```
Comment (ivanchongzhien) [inline src/main/java/seedu/duke/storage/Writer.java:3]: Same as Loader class, consider import static?

**Review by ivanchongzhien — CHANGES_REQUESTED** [review]
Mostly went through the code for Writer and Loader, skipped your tests. LGTM except maybe change the way you import constants?

File: src/main/java/seedu/duke/common/Constants.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
+package seedu.duke.common;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class Constants {
```
[the remark this answers is not in the document]
Comment (8kdesign) [reply src/main/java/seedu/duke/common/Constants.java:7]: Hmm. But some will overlap, like the index of the fields. We should try to merge them.

**Review by ivanchongzhien — APPROVED** [review]
LGTM

**Pull request merged**